### ISSUE TITLE
default number of threads should be number of physical cores, not logical

### ISSUE THREAD

**Issue by grr** [cculianu/brute38#3]
the default number of threads should be set to the number of **physical** cores, not **logical** cores. If you have a 4 core machine with hyperthreads, the default number of threads is 8. This is actually a bit slower than if 4 threads were used, since each core is already fully utilized and the extra 4 threads incur extra context switching.

**Comment by cculianu** [comment]
Yeah you're right.  I'll look at modifying this code.  Not sure if go provides a platform-neutral mechanism to detect this but I'll lookk into it (it's been a while since I programmed in go).
